libs/hdf-converters/src/msft-secure-score-mapper.ts

```diff
@@ -0,0 +1,203 @@
+import {
+  SecureScore,
+  ControlScore,
+  SecureScoreControlProfile
+} from '@microsoft/microsoft-graph-types';
+import {ExecJSON} from 'inspecjs';
+import {version as HeimdallToolsVersion} from '../package.json';
+import {BaseConverter, ILookupPath, MappedTransform} from './base-converter';
+
+export class MsftSecureScoreMapper extends BaseConverter {
+  withRaw: boolean;
+  profiles: SecureScoreControlProfile[];
+
+  private getProfiles(controlName: string): SecureScoreControlProfile[] {
+    return this.profiles.filter((profile) => profile.id === controlName);
+  }
+
+  mappings: MappedTransform<
+    ExecJSON.Execution & {passthrough: unknown},
+    ILookupPath
+  > = {
+    platform: {
+      name: 'Heimdall Tools',
+      release: HeimdallToolsVersion
+    },
+    version: HeimdallToolsVersion,
+    statistics: {},
+    profiles: [
+      {
+        name: 'Microsoft Secure Score Scan',
+        title: {
+          transformer: (data: SecureScore) =>
+            `Azure Secure Score report - TenantID: ${data.azureTenantId}`
+        },
+        supports: [],
+        attributes: [],
+        groups: [],
+        status: 'loaded',
+        controls: [
+          {
+            path: 'controlScores',
+            id: {
+              transformer: (data: ControlScore) =>
+                `${data.controlCategory}:${data.controlName}`
+            },
+            title: {
+              transformer: (data: ControlScore) => {
+                const titles = this.getProfiles(data.controlName || '')
+                  .filter((profile) => profile.title !== undefined)
+                  .map((profile) => profile.title);
+
+                if (titles.length > 0) {
+                  return titles.join('... ');
+                } else {
+                  return `${data.controlCategory || ''}:${data.controlName || ''}`;
+                }
+              }
+            },
+            desc: {path: 'description'},
+            impact: {
+              transformer: (data: ControlScore) => {
+                // return controlCategory from the profile document where its id matches the controlName
+                const knownMaxScores = this.getProfiles(
+                  data.controlName || ''
+                ).map((profile) => profile.maxScore || 0);
+
+                if (knownMaxScores.length === 0) {
+                  return 0.5;
+                }
+
+                const highMaxScore = Math.max(...knownMaxScores);
+                return highMaxScore / 10.0;
+              }
+            },
+            refs: [],
+            tags: {
+              group: {
```

Comment:
why are we using a different naming scheme than the original format?  shouldn't we just call this 'controlcategory' or 'category'?

Reply:
updated to `category`